src/Spice86.Core/Emulator/VM/Machine.cs

```diff
@@ -346,4 +349,12 @@ public sealed class Machine : IDisposable {
         Dispose(disposing: true);
         GC.SuppressFinalize(this);
     }
+
+    public void Accept(IEmulatorDebugger emulatorDebugger) {
+        Cpu.Accept(emulatorDebugger);
+        ((IDebuggableComponent)VgaCard).Accept(emulatorDebugger);
```

Comment:
Is cast needed? It seems VgaCard implements IDebuggableComponent isnt it?

Reply:
I don't like it either (too many parentheses, makes the code harder to read), however the types are declared as IVideoCard, IMemory and such, that's why the cast is needed.

JetBrains Rider doesn't detect unneeded casts.

Reply:
Maybe the IVideoCard, IMemory etc... interfaces could inherit from the IDebuggableComponent interface ?

Reply:
Yes that would be the best solution